openhcl/virt_mshv_vtl/src/processor/snp/mod.rs

```diff
@@ -1449,85 +1455,111 @@ impl<T: CpuIo> X86EmulatorSupport for UhEmulationState<'_, '_, T, SnpBacked> {
         self.vp.partition.caps.vendor
     }
 
-    fn state(&mut self) -> Result<x86emu::CpuState, Self::Error> {
+    fn gp_sign_extend(&mut self, reg: usize) -> i64 {
+        self.gp(reg) as i64
```

Comment:
I'm not sure this works?

Reply:
It definitely does not, in fact there shouldn't be any sign extension at the emulator support level. Conversion is done in the implementation of the cpu trait